Pont 3 api key to constructor

pont-us opened the pull request:
This PR adds `cds_api_url` and `cds_api_key` parameters to the `CDSDataOpener` and `CDSDataStore` constructors. Previously, the CDS API URl and key could only be specified via environment variables or a configuration file.

TonioF reviewed (changes requested):
The tests work nicely. However, I found it a bit hard to figure out the differences between the three new tests 'test_client_url_and_key_parameters', 'test_client_url_and_key_environment_variables' and 'test_client_url_and_key_rc_file'. I recommend that you at least document the tests a bit so that we know why e.g. to test the file access you pass only the client_class to the opener.
Also, it seems that the tests do not consider any preexisting environment variables and may overwrite those. This should be prevented.
Finally, I'd recommend to create a new TestClass for the new tests (e.g., ClientUrlTest) and use setUp()- and tearDown()-functions instead of the two static methods you have used.

pont-us commented:
@TonioF Thanks for the suggestions -- all implemented. I've put the new tests into a new `ClientUrlTest` class, added docstrings, added `setUp` and `tearDown` functions, and done some other refactoring to remove duplication and improve code readability. Existing environment variables are now properly saved and restored.
